statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,254 @@
+        if self.endog.ndim == 1:
+            raise ValueError("Only gave one variable to VECM")
+        self.y = self.endog #TODO delete this line if y not necessary
+        self.neqs = self.endog.shape[1]
+
+    def predict(self, params, start=None, end=None, lags=1, trend='c'):
+        """
+        Returns in-sample predictions or forecasts
+        """#TODO: docstring + implementation
+        pass
+
+    def fit(self, max_diff_lags=None, method='ml', ic=None, 
+            deterministic_terms='', verbose=False, coint_rank=None):
+        """
+        Fit the VECM
+
+        Parameters
+        ----------
+        max_diff_lags : int
+            Maximum number of lags to check for order selection, defaults to
+            12 * (nobs/100.)**(1./4), see select_order function ##### ##### ##### #####
+        method : {'ls', 'egls', 'ml'}
+            Estimation method to use.
+        ic : {'aic', 'fpe', 'hqic', 'bic', None}
+            Information criterion to use for VECM order selection.
+            aic : Akaike
+            fpe : Final prediction error
+            hqic : Hannan-Quinn
+            bic : Bayesian a.k.a. Schwarz
+        deterministic terms, str {"", "c", "lt", "s"}
+            "" - no deterministic terms
+            "c" - constant
+            "lt" - linear trend
+            "s" - seasonal terms
+            Combinations of these are possible (e.g. "clt" for linear trend 
+            with intercept)
+
+        Notes
+        -----
+        Lutkepohl pp. 146-153
+
+        Returns
+        -------
+        est : VARResults
+        """#TODO: docstring + implementation
+        
+        #TODO: trend
+        
+        # select number of lags (=p-1)
+        if ic is None:
+            if max_diff_lags is None:
+                diff_lags = 1
+            else:
+                diff_lags = max_diff_lags
+        else:
+            selections = self.select_order(max_diff_lags=max_diff_lags, verbose=verbose)
+            if ic not in selections:
+                raise Exception("%s not recognized, must be among %s"
+                                % (ic, sorted(selections)))
+            diff_lags = selections[ic]
+            if verbose:
+                print('Using %d based on %s criterion' %  (diff_lags, ic))
+        
+        # estimate parameters
+        if method=='ls':
+            return self._estimate_vecm_ls(diff_lags, deterministic_terms)
+        elif method == 'egls':
+            if coint_rank is None:
+                coint_rank = 1
+            return self._estimate_vecm_egls(diff_lags, deterministic_terms, coint_rank)
+        elif method == 'ml':
+            if coint_rank is None:
+                coint_rank = 1
+            return self._estimate_vecm_ml(diff_lags, deterministic_terms, coint_rank)
+        else:
+            raise Exception("%s not recognized, must be among %s"
+                            % (method, ("ls", "egls", "ml")))
+        
```

Comment:
The are three different returns depending on which method for estimation shall be used (LS, EGLS, or ML).